packages/api/src/controllers/clip.ts

```diff
@@ -61,49 +58,35 @@ app.use(
 );
 
 async function getProcessingClipsByRequesterId(
-  requesterId: string
-): Promise<WithID<Asset>[]> {
-  const createdAfter = Date.now() - 10 * 60 * 1000;
-  const assets = await db.asset.find([
-    sql`data->'status'->>'phase' IN ('waiting', 'processing', 'running')`,
-    sql`data->'source'->>'requesterId' = ${requesterId}`,
-    sql`coalesce((data->>'createdAt')::bigint, 0) > ${createdAfter}`,
+  requesterId: string,
+  ownerId: string
+): Promise<WithID<Task>[]> {
+  const scheduledAt = Date.now() - 5 * 60 * 1000;
+  const [tasks] = await db.task.find([
+    sql`data->>'userId' = ${ownerId}`,
+    sql`coalesce((data->>'scheduledAt')::bigint, 0) > ${scheduledAt}`,
   ]);
```

Comment:
Can we have that as a dedicated query function on `task-table`? Like the other rate limiting queries, with a clear name so we can easily reason about in all the rate-limiting logics we create all around. e.g. this is probably just `getPendingClipTasks` or something

Reply:
Right! I moved it in the task table and made the `clipType` as optional parameter, so it can used not just for clips (`getProcessingTaskByRequesterId`)